Makefile

```diff
@@ -447,7 +447,7 @@ check_ford_install:
 	@echo "Ford command $(FORD) not in path -- is it installed?\\n\\tConsider installing with 'pip install --user ford' and add ${HOME}/.local/bin to PATH" ; which $(FORD)
 endif
 
-GIT_VERSION := $(shell git describe --tags --long --match "v*" --first-parent HEAD)
+GIT_VERSION := $(shell git describe --tags --always --long --match "v*" --first-parent HEAD)
```

Comment:
Probably want to also drop `--long` too, so that when exactly on a tag, the version doesn't include the commit etc

Reply:
Perhaps this is the right time to solve something that I've been trying to do up to this point unsuccessfully: when stella runs, it currently prints out the version number, but I think it would be useful to print out the (truncated?) commit number as well.  My idea was to use define the macro GIT_VERSION using the above and then everything should work. The problem is if, say, you 'git pull' and stella.f90(pp) is not modified, then using 'make' won't necessarily make the change in the right place.  The alternative is to always build stella.f90, but then this will mess with people who compulsively 'make' just to ensure things have been properly built. Is this something that can be done in a streamlined way such that the user doesn't even notice it happens?

Reply:
Ah, I do actually have a solution for this, which ensures the version number (including commit hash) is always up-to-date and correct: https://github.com/PlasmaFAIR/Fortran-git-version
Originally written for gs2, I've pulled it out specifically to make it easier to integrate into other projects :)

I can make a PR for this. It's a bit nicer using CMake, but does still work with plain makefiles.

Reply:
I think @ZedThree may be well placed to address this. He has done some similar work for GS2 recently.